redaxo/src/addons/backup/lib/backup.php

```diff
@@ -83,15 +94,42 @@ public static function importDb($filename)
             return $return;
         }
 
+        $decompressedFilename = false;
+
+        if ('gz' === rex_file::extension($filename)) {
+            $compressor = new rex_backup_file_compressor();
+            $decompressedFilename = $compressor->gzDeCompress($filename);
```

Comment:
Eigentlich fände ich es schöner, direkt das gz-file einzulesen. Also ganz ohne temp-datei.
könnte auch eine methode im compressor sein, nur dass die halt nicht in eine datei schreibt, sondern direkt den inhalt zurückliefert. (die datei wird ja sowieso als ganzes am ende eingelesen)

Dann würde das aufräumen der temp-datei auch entfallen.

Reply:
ich wollte nicht die datei vollständig in den speicher laden daher hab ich es so gemacht.

ausserdem ist es in dieser form auch symetrisch mit gzCompress

Reply:
Die Datei landet danach aber sowieso vollständig im Speicher. Sie wird ja über `rex_file::require` dann anschließend eingelesen.

Reply:
aber nach deinem vorgehen würden wir 2 dateien in den speicher laden.

an der bisherigen logik (bzgl. file-loading) hab ich/wollte ich hier nix ändern.

Reply:
ohh ich glaub ich verstehe jetzt was du meinst. ich schau mal

Reply:
also als pseudo-code meinte ich:

```
if (gz)
  $conts = compressor->content
else
  $conts = rex_file::require
```

Reply:
jop, hab es angepasst